File: Chapter5/5.7/power/cpp/power.cpp
#include <iostream>

unsigned power(unsigned x, unsigned y){
  unsigned result = 1;
  unsigned weight = x;
  while (y != 0) {
    if ( (y&1) == 1){
      result *= weight;      
    }
    y >>= 1;
    weight *= weight;
  }
  return result;
}

int main() {
  int x,y;
  std::cin>>x;std::cin>>y;
  std::cout<<"x^y:"<<power(x,y)<<std::endl;
  return 0;
}

File: Chapter5/5.7/power/java/power.java
import java.util.Scanner;

public class power{
    public static int pow(int x, int y){
	int result = 1;
	int weight = x;
	while (y != 0) {
	    if ((y&1)==1) result *= weight;
	    weight *= weight;
	    y >>= 1;
	}
	return result;
    }

    public static void main(String[] args){
	Scanner in = new Scanner(System.in);
	int x=in.nextInt(), y=in.nextInt();
	if ((x<0) || (y<0)) return;
	System.out.println("x^y:"+pow(x,y));
    }
}